### ISSUE TITLE
Make Image Gradient display complete asset path, like Gradient Baker

### ISSUE THREAD

**Issue by HogJonny-AMZN** [o3de/o3de#13238]
**Is your feature request related to a problem? Please describe.**
The Gradient Baker will display a more complete asset path then the Image Gradient component does.

**Describe the solution you'd like**
I would like the Image Gradient asset path to display as much information as the Gradient Baker does.

**Additional context**
In this example, I am baking a gradient into folder nested under the current level:
\Levels\OK-Delete\bakes\baked_output_gsi.png
![image](https://user-images.githubusercontent.com/23222931/202276919-793e83d6-232a-415e-b3ec-f276df4dab3b.png)

This Image Gradient is using the same location, but only displays the asset images name:
![image](https://user-images.githubusercontent.com/23222931/202278069-75993c74-1958-4e68-8570-4c98e478ce20.png)

**Comment by cgalvan** [comment]
@HogJonny-AMZN It looks like the images in the description are pointed at different assets, but I see what you are saying. In this example, the Image Gradient asset path will display `baked_output_gsi.png` if it was pointed at the same asset as the Gradient Baker on top.

The difference would be easy to fix. The reason it shows up differently (even though the property controls look similar) is because the Gradient Baker actually stores a source asset path, which displays the asset directory relative path in the control. Whereas the Image Gradient actually stores an asset data type, and its display only shows the source file name, instead of the relative path.

It would be straightforward to change what is displayed, but its worth noting it wouldn't just change it for the Image Gradient, it would be for all asset fields, of which there are many. @rainbj @amzn-leenguy do y'all have thoughts on this?

**Comment by HogJonny-AMZN** [comment]
yes, these are different control patterns ... but when I wrote this, I wasn't in that headspace, the disparity reared it's head cognitively while 'in the flow.

Gradient Baker: Output Path, is jusy a location to generate a image file, it is not a standard asset feild.
Image Gradient: Image Asset, is a standard asset path.

I'm only suggesting this ticket, as the more similar they are in UX, reduces some cognitive load while using them both in a level workflow.

**Comment by cgalvan** [comment]
Yeah I actually agree that for consistency we should have them both show the relative path. I only called out the different property types to explain why they behaved differently, because even though they are different property types, they both use a sub-class of the same property asset control for the UI.

**Comment by amzn-leenguy** [comment]
@HogJonny-AMZN and @cgalvan , thanks for tagging me and bringing this to my attention:

I can see why the UX pattern here is showing the path, in relative to the other one just showing the asset; since the label mentions "path" and "asset". but to focus on the consistency part of this, I agree that we should come to a standard for both and apply this to all types.

Questions:
- It seems in the comments previous; You both are suggesting to show the complete full path on all fields that align this this scenario (E.g. file picking something, it shows the full path)

- Cognitively, is it useful for a user to see the complete path location still lingering there, after they have just chosen it from the file picker? Same thing, if a user chose it in the past, and they are scrolling through these component cards weeks later, is it useful to see the complete path there all the time? 

I guess Im asking to weigh the advantages vs disadvantages here of showing more information than needed. The three options i can see here are:

\Levels\OK-Delete\bakes\baked_output_gsi.png
\Levels\...\baked_output_gsi.png
baked_output_gsi.png

If we make it all the fields show the full path all the time (as suggested above), is the cognitive load points worth it? Because we only get so many points at the end of the day. 

I do not have any preferences here for full path or not, but I just want to have a discussion on what is on top of mind for me.

**Comment by rainbj** [comment]
Thanks for the tag @cgalvan. A couple small call-outs based on what I'm seeing here. 
1) Is this baking process recursive? Meaning, the user is having to re-"bake" over and over again. If so, we should follow the new patterns I have been working with @mbalfour-amzn with the landscape tool (Auto save). I think you can see it in your second image above. He and I had worked on this auto save process which streamlined the file output instead of manually saving it over and over again. So file name and extension is less important as the file location.  
2) "Output path" generally is just a path and not also a file extension. We might want to just call it "file" or "asset". 
3) Which is, to path or not to path. That is the question :) My answer would be, is having the path useful in the primary use case? Meaning, Is it critical for the user to know this path location to work with this file.  My expectation is that knowing the file location is not important as a primary use case, and only important as a secondary use case. So we could consider maybe adding the full url to a tooltip or some other method for the user to see this data (on the folder button?).  Showing an entire URL is generally distracting. So I would vote to only show full path when it's absolutely useful.

**Comment by mbalfour-amzn** [comment]
So, just to throw a bit more in the mix to think about here. In _general_, when we have asset references on component cards, they're referring to existing assets that we're loading and using. 

In this particular case, with Image Gradient, we're referring to an asset that we're both loading _and potentially saving_, and not even necessarily saving from the same component. (i.e. the asset could either be saved from the Image Gradient editing or from the Gradient Baker)

I believe the cognitive value difference that's coming into play here is that it's generally more important to know the full path when you're dealing with both loading and saving than when you're only dealing with loading.

So, the things to consider here:
1. The Image Gradient is sometimes using images that are only loaded, and sometimes using images that are both loaded and saved in the same workflow.
2. Showing the full path on the asset will show it for both workflows.
3. Hiding the full path on the asset (like we do now) will hide it for both workflows.
4. If we modify the asset reference widget to show the full path, would we want that change to affect all components, or just this one? If it's just this one, is the inconsistency OK?

**Comment by mbalfour-amzn** [comment]
One more thing to think about - I believe @rainbj has also requested that we add a collapsible display of image stats (resolution, format) to the Image Gradient. If we do that, maybe the full path should go in there?

**Comment by cgalvan** [comment]
Just to make sure we are clear on the verbiage, I'm not proposing to show the full path, which would be the absolute path (e.g. C:\my\path\to\project\textures\image_gsi.png).

I was proposing to show the relative path (e.g. textures\image_gsi.png) all the time for assets instead of only the filename for consistency. I can also see though how that might be distracting if you have a really deep relative path.

It seems like at a minimum we should either:
A. Have the Image Gradient show the relative path when it's source type is "Use Existing Image"  OR
B. Add the collapsible stats to the Image Gradient that @mbalfour-amzn mentioned and include the full path there

Would B have made your workflow seamless @HogJonny-AMZN ? Or would the fact that one is showing a relative path and one is only showing the filename still be confusing?

**Comment by HogJonny-AMZN** [comment]
First let me just iterate that cognitively I was just in a workflow, 'working with images', with two different components but there was an itch because things felt disjointed.  I don't know what the perfect answer is, just that it should probably be consistent?

I think where I am at after the thread is:
- by default use the short path, display the image file name.ext
- allow a hover tooltip, to display the longer relative path (not abs path)
-  Add the collapsible stats to the Image Gradient that @mbalfour-amzn mentioned and include the full path there

**Comment by HogJonny-AMZN** [comment]
From a scaled production standpoint, and in consideration of things like re-org of data during mid-production, or troubleshooting during a time critical point in production (like a release window):

- Anywhere a path is displayed, I should probably have an easy way to copy that path

- If we display the full path in a collapsible, but that field is not tied to a file browser, then I should have something like a right-click action, to copy that path

That would mean the UX is not making me hope around, back to the other area of the card, to get to a file browser, so I can then copy the path.  Keep me in the flow and context of the information.

- I found the path
- I see the path
- that is the path I may need to utilize

**Comment by cgalvan** [comment]
> From a scaled production standpoint, and in consideration of things like re-org of data during mid-production, or troubleshooting during a time critical point in production (like a release window):
> 
> * Anywhere a path is displayed, I should probably have an easy way to copy that path
> * If we display the full path in a collapsible, but that field is not tied to a file browser, then I should have something like a right-click action, to copy that path
> 
> That would mean the UX is not making me hope around, back to the other area of the card, to get to a file browser, so I can then copy the path. Keep me in the flow and context of the information.
> 
> * I found the path
> * I see the path
> * that is the path I may need to utilize

It might be nice to have a "copy to clipboard" button that is pretty common for things similar to this, that would copy the absolute path for the source asset.

**Comment by HogJonny-AMZN** [comment]
> > From a scaled production standpoint, and in consideration of things like re-org of data during mid-production, or troubleshooting during a time critical point in production (like a release window):
> > 
> > * Anywhere a path is displayed, I should probably have an easy way to copy that path
> > * If we display the full path in a collapsible, but that field is not tied to a file browser, then I should have something like a right-click action, to copy that path
> > 
> > That would mean the UX is not making me hope around, back to the other area of the card, to get to a file browser, so I can then copy the path. Keep me in the flow and context of the information.
> > 
> > * I found the path
> > * I see the path
> > * that is the path I may need to utilize
> 
> It might be nice to have a "copy to clipboard" button that is pretty common for things similar to this, that would copy the absolute path for the source asset.

Yes, that would be very useful.
